pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/BitSlicedRangeIndexReader.java

```diff
@@ -45,30 +50,34 @@ public BitSlicedRangeIndexReader(PinotDataBuffer dataBuffer) {
     _min = dataBuffer.getLong(offset);
     offset += Long.BYTES;
     _offset = offset;
+    _max = metadata.hasDictionary()
```

Comment:
Suggest making them part of the header so that the index is self-contained. It will be a backward-incompatible change, but since we haven't released it yet, it should be fine.

Reply:
Isn't it better to rely on the metadata? If information were stored in two places it can be out of sync. This way the column metadata is the source of truth.

Reply:
Here using either of them should be fine, especially when the creator is using the min/max value from the column metadata.
In some places reading the metadata min/max value might cause precision loss because it involves ser/de of the values.